### PR TITLE
Migrate to OpenTracing Java 0.31.0.RC1

### PR DESCRIPTION
**Pull request by objectiser**

### CODE REVIEW

**Comment by objectiser** [comment]
Jax-rs instrumentation is dependent upon this, so thought I would get this one done aswell.

It shows up a potential issues - the `TracedRunnable` and `TracedCallable` both start a scope in the new thread, but have to decide whether to 'finishOnClose'.

I've set to false for now, as potentially other instrumentation should control that decision - but in the original continuation approach control for managing the completion/finish of a span could have been delegated through a series of exec'd runnables etc.

File: src/main/java/io/opentracing/contrib/concurrent/TracedCallable.java [hunk 1]
```diff
@@ -11,20 +12,24 @@
 public class TracedCallable<V> implements Callable<V> {
 
   private final Callable<V> delegate;
-  private final Continuation continuation;
+  private final Span continuation;
+  private final Tracer tracer;
 
-  public TracedCallable(Callable<V> delegate, ActiveSpan activeSpan) {
+  public TracedCallable(Callable<V> delegate, Tracer tracer) {
```
Comment (jpkrohling) [inline src/main/java/io/opentracing/contrib/concurrent/TracedCallable.java]: Is there a way to keep a backwards compatible version available, perhaps with a `@Deprecated` tag?

File: src/main/java/io/opentracing/contrib/concurrent/TracedRunnable.java [hunk 2]
```diff
@@ -1,29 +1,33 @@
...
  */
 public class TracedRunnable implements Runnable {
 
   private final Runnable delegate;
-  private final Continuation continuation;
+  private final Span continuation;
+  private final Tracer tracer;
 
-  public TracedRunnable(Runnable delegate, ActiveSpan activeSpan) {
+  public TracedRunnable(Runnable delegate, Tracer tracer) {
```
Comment (jpkrohling) [inline src/main/java/io/opentracing/contrib/concurrent/TracedRunnable.java]: Same as the other comment about backwards compatibility

**Review by jpkrohling — COMMENTED** [review]
Once the backwards compatibility issue is sorted out, this can be merged.

File: src/main/java/io/opentracing/contrib/concurrent/TracedCallable.java (hunk 1 above)
Comment (objectiser) [reply src/main/java/io/opentracing/contrib/concurrent/TracedCallable.java]: Backward compatibility would be impossible - the `ActiveSpan` class no longer exists.

Comment (jpkrohling) [reply src/main/java/io/opentracing/contrib/concurrent/TracedCallable.java]: Good point :)

File: src/main/java/io/opentracing/contrib/concurrent/TracedCallable.java [hunk 3]
```diff
@@ -11,20 +12,24 @@
...
     this.delegate = delegate;
-    this.continuation = activeSpan != null ? activeSpan.capture() : NoopContinuation.INSTANCE;
+    this.tracer = tracer;
+    this.continuation = tracer.scopeManager().active() == null ? null : tracer.scopeManager().active().span();
```
Comment (pavolloffay) [inline src/main/java/io/opentracing/contrib/concurrent/TracedCallable.java]: I think it's cleaner to use `NoopScope` like it was before.

**Review by jpkrohling — APPROVED** [review]

File: src/main/java/io/opentracing/contrib/concurrent/TracedCallable.java (hunk 3 above)
Comment (objectiser) [reply src/main/java/io/opentracing/contrib/concurrent/TracedCallable.java]: Its not the scope that is being passed across threads, it is the span.

Comment (pavolloffay) [reply src/main/java/io/opentracing/contrib/concurrent/TracedCallable.java]: could you rename continuation then to something more accurate

**Comment by objectiser** [comment]
@pavolloffay Updated PR - also created branch and updated publish.sh script. Ready to merge/release?

**Comment by pavolloffay** [comment]
+1 LGTM

**Pull request merged**